src/wallet/Wallet.ts

```diff
@@ -0,0 +1,2192 @@
+		outputType: OutputType = DEFAULT_OUTPUT,
+		config?: MintProofsConfig,
+	): Promise<Proof[]> {
+		const { privkey, keysetId, proofsWeHave } = config ?? {};
+		if (amount <= 0) {
+			this._logger.warn('Invalid mint amount: must be positive', { amount });
+			throw new Error('Amount must be positive');
+		}
+		// Create outputs for our proofs (we are receiving, so no includeFees)
+		const keyset = this.keyChain.getKeyset(keysetId);
+		const outputs = this.configureOutputs(amount, keyset, outputType, false, proofsWeHave);
+		const blindedMessages = outputs.map((d) => d.blindedMessage);
+		let mintPayload: MintPayload;
+		if (typeof quote === 'string') {
+			mintPayload = {
+				outputs: blindedMessages,
+				quote: quote,
+			};
+		} else {
+			if (!privkey) {
+				const message = 'Can not sign locked quote without private key';
+				this._logger.error(message);
+				throw new Error(message);
+			}
+			const mintQuoteSignature = signMintQuote(privkey, quote.quote, blindedMessages);
+			mintPayload = {
+				outputs: blindedMessages,
+				quote: quote.quote,
+				signature: mintQuoteSignature,
+			};
+		}
+		let signatures;
+		if (method === 'bolt12') {
+			({ signatures } = await this.mint.mintBolt12(mintPayload));
+		} else {
+			({ signatures } = await this.mint.mint(mintPayload));
+		}
+		if (signatures.length !== outputs.length) {
+			const message = `Mint returned ${signatures.length} signatures, expected ${outputs.length}`;
+			this._logger.error(message);
+			throw new Error(message);
+		}
+		this._logger.debug('MINT COMPLETED', { amounts: outputs.map((o) => o.blindedMessage.amount) });
+		return outputs.map((d, i) => d.toProof(signatures[i], keyset));
+	}
+
+	/**
+	 * Melt proofs for a given melt quote created with the bolt11 or bolt12 method.
+	 *
+	 * @remarks
+	 * Creates NUT-08 blanks (1-sat) for Lightning fee return. Get these by setting a
+	 * config.onChangeOutputsCreated callback for async melting. @see completeMelt.
+	 * @param method Payment method of the quote.
+	 * @param meltQuote The bolt11 or bolt12 melt quote.
+	 * @param proofsToSend Proofs to melt.
+	 * @param outputType Proof generation config (random, deterministic, p2pk, etc.).
+	 * @param config Optional (keysetId, onChangeOutputsCreated).
+	 * @returns MeltProofsResponse.
+	 * @throws If params are invalid or mint returns errors.
+	 * @see https://github.com/cashubtc/nuts/blob/main/08.md.
+	 */
+	private async _meltProofs<T extends 'bolt11' | 'bolt12'>(
+		method: T,
+		meltQuote: T extends 'bolt11' ? MeltQuoteResponse : Bolt12MeltQuoteResponse,
+		proofsToSend: Proof[],
+		outputType: OutputType = DEFAULT_OUTPUT,
+		config?: MeltProofsConfig,
+	): Promise<MeltProofsResponse> {
+		const { keysetId, onChangeOutputsCreated } = config || {};
+		const keyset = this.keyChain.getKeyset(keysetId);
+		const feeReserve = sumProofs(proofsToSend) - meltQuote.amount;
+		let outputData: OutputDataLike[] = [];
+
```

Comment:
Should we throw if sumProofs is less than meltQuote amount (v2 did not)?
Or is a partial melt ever ok?